src/db.c

```diff
@@ -31,18 +32,20 @@ int execute_db_operations(void)
 			free(available_dbs);
 			return ret;
 		}
+
+		if (db_ops_counter + 1 < MAX_AVAILABLE_DBS) {
+			db_ops_counter++;
+		}
```

Comment:
Remove, it is too late in the scope to do this.

Reply:
I believe It shouldnt be the responsibility of the init function to incriment the counter.
The counter is created and should be managed by the `db` file no?

Also its better for designing the new database implementation to not have to think about managing the 
counter by incrementing it and checking if its out of bounds, i believe this is something that should be done as a different procedure and not in every `init_func` for each database system we implement

Reply:
A better way to do it is:
1. Check if counter is out of bounds
2. If not, run init function
3. If it is, exit

Reply:
Ok sure. Where should the increment happen though? We could do it after successfully running the init function.

Reply:
Yes, run init -> increment